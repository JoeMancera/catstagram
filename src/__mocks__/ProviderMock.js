/* eslint-disable react/destructuring-assignment */
import 'jsdom-global/register';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

const ProviderMock = (props) => (
  <BrowserRouter history={history}>
    {props.children}
  </BrowserRouter>
);

export default ProviderMock;
