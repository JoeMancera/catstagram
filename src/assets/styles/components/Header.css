.title{
  display: none;
}

.menu{
  background-color: var(--primary);
  height: 45px;
  width: 100vw;
  position: fixed;
  bottom: 0;
  z-index: 1;
}

.menu ul{
  display: flex;
  justify-content: space-around;
  list-style: none;
  padding-inline-start: 0px;
  margin: 0;
  padding-top: 12px;
}

.menu ul a{
  text-decoration: none;
  color: var(--withe);
}

.menu-active{
  color: var(--secondary);
}

.menu ul figure{
  margin: 0;
}

.nav-text {
  display: none;
}

.nav-item:hover {
  color: var(--secondary);
  font-size: 1.1rem;
}

@media only screen and (min-width: 540px){
  .hero-image{
    -webkit-filter: drop-shadow(0 0 1px #222);
    filter: drop-shadow(0 0 1px #222);
    max-width: 32px;
    margin-top: 3px;
  }

  .title{
    display: inline;
    margin-left: 8px;
  }

  .title h1{
    display: none;
  }

  .menu{
    justify-content: space-between;
    display: flex;
    top: 0;
  }

  .menu ul{
    padding-inline-start: unset;
    min-width: 90%;
  }
}

@media only screen and (min-width: 768px){
  .cat_image_horizontal{
    width: 460px;
    height: 270px;
  }
  
  .cat_image_vertical{
    width: 460px;
    height: 490px;
  }

  .nav-text {
    margin: 0;
    margin-left: 8px;
    display: block;
  }

  .nav-item {
    display: flex;
  }
}