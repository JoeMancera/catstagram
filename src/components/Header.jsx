import React from 'react';
import { Link } from 'react-router-dom';

import '../assets/styles/components/Header.css';
import home from '../assets/static/home.png';
import search from '../assets/static/search.png';
import favorite from '../assets/static/favorite.png';
import profile from '../assets/static/user.png';
import logo from '../assets/static/Icon.png';

const Header = () => (
  <header className='header'>
    <section>
      <nav className='menu' aria-label='main-navigation'>
        <span className='title'>
          <img className='hero-image' src={logo} alt='Catstagram logo' />
          <h1>Catstagram</h1>
        </span>
        <ul>
          <li className='menu-active'>
            <Link className='nav-item' to='/'>
              <figure>
                <img src={home} alt='Home icon' />
              </figure>
              <p className='nav-text'>Home</p>
            </Link>
          </li>
          <li>
            <Link className='nav-item' to='/search'>
              <figure>
                <img src={search} alt='Search icon' />
              </figure>
              <p className='nav-text'>Search</p>
            </Link>
          </li>
          <li>
            <Link className='nav-item' to='/favorites'>
              <figure>
                <img src={favorite} alt='Search icon' />
              </figure>
              <p className='nav-text'>Favorites</p>
            </Link>
          </li>
          <li>
            <Link className='nav-item' to='/profile'>
              <figure>
                <img src={profile} alt='User icon' />
              </figure>
              <p className='nav-text'>Profile</p>
            </Link>
          </li>
        </ul>
      </nav>
    </section>
  </header>
);

export default Header;
