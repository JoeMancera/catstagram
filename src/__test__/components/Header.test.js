import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Header from '../../components/Header';

describe('Tests for <Header /> component', () => {

  test('Header render test,should return 1', () => {
    const header = shallow(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.length).toEqual(1);
  });

  test('Header Title render test, should return catstagram', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );

    expect(header.find('.title').text()).toEqual('Catstagram');
  });
});

describe('<Header /> Snapshot test', () => {
  test('Should match the generated snapshot', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );

    expect(header.toJSON()).toMatchSnapshot();
  });

});
