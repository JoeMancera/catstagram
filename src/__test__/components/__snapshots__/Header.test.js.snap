// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot test Should match the generated snapshot 1`] = `
<header
  className="header"
>
  <div
    className="title"
  >
    <h1>
      <img
        alt="Catstagram logo"
        className="hero-image"
        src="test-file-stub"
      />
      Catstagram
    </h1>
  </div>
  <section>
    <nav
      aria-label="main-navigation"
      className="menu"
    >
      <ul>
        <li
          className="menu-active"
        >
          <a
            className="nav-item"
            href="/"
            onClick={[Function]}
          >
            <figure>
              <img
                alt="Home icon"
                src="test-file-stub"
              />
            </figure>
          </a>
        </li>
        <li>
          <a
            className="nav-item"
            href="/search"
            onClick={[Function]}
          >
            <figure>
              <img
                alt="Search icon"
                src="test-file-stub"
              />
            </figure>
          </a>
        </li>
        <li>
          <a
            className="nav-item"
            href="/favorites"
            onClick={[Function]}
          >
            <figure>
              <img
                alt="Search icon"
                src="test-file-stub"
              />
            </figure>
          </a>
        </li>
        <li>
          <a
            className="nav-item"
            href="/profile"
            onClick={[Function]}
          >
            <figure>
              <img
                alt="User icon"
                src="test-file-stub"
              />
            </figure>
          </a>
        </li>
      </ul>
    </nav>
  </section>
</header>
`;
